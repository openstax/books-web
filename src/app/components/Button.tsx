import styled from 'styled-components';
import theme, { ColorSet } from '../theme';
import { textRegularStyle } from './Typography';

const applyColor = (color: ColorSet) => `
  color: ${color.foreground};
  background-color: ${color.base};

  :hover {
    background-color: ${color.darker};
  }
  :active {
    background-color: ${color.darkest};
  }
`;

export default styled.button<{variant: undefined | 'primary' | 'secondary' | 'default'}>`
  align-items: center;
  border: none;
  border-radius: 2px;
  display: flex;
  justify-content: center;
  text-align: center;
  box-sizing: border-box;
  cursor: pointer;
  display: inline-flex;
  height: 5rem;
  margin: 0;
  padding: 0 2rem;
  text-decoration: none;
  user-select: none;
  white-space: nowrap;
  width: auto;

  ${textRegularStyle}
  font-weight: bold;

  ${(props) => props.variant === 'primary' && applyColor(theme.color.primary.orange)}
  ${(props) => props.variant === 'secondary' && applyColor(theme.color.secondary.lightGray)}
  ${(props) => props.variant === 'default' && applyColor(theme.color.neutral)}
`;

// tslint:disable-next-line:variable-name
export const ButtonGroup = styled.div`
  display: grid;
  grid-gap: 1rem;
  max-width: 100%;
`;
