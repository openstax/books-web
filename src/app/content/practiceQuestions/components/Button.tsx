import styled from 'styled-components/macro';
import { PlainButton } from '../../../components/Button';
import theme from '../../../theme';

interface ButtonProps {
  withoutBackground: boolean;
}

export default styled(PlainButton)`
  backgrouond-color: ${theme.color.primary.orange.base};
  color: ${theme.color.text.white};
  font-size: 1.6rem;
  font-weight: bold;
  height: 5rem;
  padding: 0 3rem;
  ${(props: ButtonProps) => {
    if (props.withoutBackground) {
      return `
        backgrouond-color: transparent;
        color: #027eb5;
      `;
    }
  }}
`;
