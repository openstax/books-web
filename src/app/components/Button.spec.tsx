import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';
import { ButtonLink } from './Button';

describe('Button', () => {
  it('matches snapshot - link', () => {
    const component = renderer.create(<Button component={<a href='/' />}>this is a link</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - small', () => {
    const component = renderer.create(<Button size='small' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - medium', () => {
    const component = renderer.create(<Button size='medium' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - large', () => {
    const component = renderer.create(<Button size='large' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - primary', () => {
    const component = renderer.create(<Button variant='primary' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - secondary', () => {
    const component = renderer.create(<Button variant='secondary' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot - default', () => {
    const component = renderer.create(<Button variant='default' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ButtonLink', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<ButtonLink />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot (decorated)', () => {
    const component = renderer.create(<ButtonLink decorated />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot (decorated and disabled)', () => {
    const component = renderer.create(<ButtonLink decorated disabled />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
