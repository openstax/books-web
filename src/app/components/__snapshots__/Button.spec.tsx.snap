// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot - default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c0:hover {
  background-color: #fafafa;
}

.c0:active {
  background-color: #e5e5e5;
}

<button
  className="c0"
/>
`;

exports[`Button matches snapshot - large 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  font-size: 1.6rem;
  height: 5rem;
  padding: 0 3rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c0:hover {
  background-color: #fafafa;
}

.c0:active {
  background-color: #e5e5e5;
}

<button
  className="c0"
/>
`;

exports[`Button matches snapshot - medium 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c0:hover {
  background-color: #fafafa;
}

.c0:active {
  background-color: #e5e5e5;
}

<button
  className="c0"
/>
`;

exports[`Button matches snapshot - primary 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #fff;
  background-color: #f36b32;
  font-weight: bold;
  border: none;
}

.c0:hover {
  background-color: #e96128;
}

.c0:active {
  background-color: #df571e;
}

<button
  className="c0"
/>
`;

exports[`Button matches snapshot - secondary 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  min-width: 12rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #fff;
  background-color: #959595;
  font-weight: bold;
  border: none;
}

.c0:hover {
  background-color: #8b8b8b;
}

.c0:active {
  background-color: #818181;
}

<button
  className="c0"
/>
`;

exports[`Button matches snapshot - small 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  font-size: 1.4rem;
  height: 3rem;
  padding: 0 2rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c0:hover {
  background-color: #fafafa;
}

.c0:active {
  background-color: #e5e5e5;
}

<button
  className="c0"
/>
`;
